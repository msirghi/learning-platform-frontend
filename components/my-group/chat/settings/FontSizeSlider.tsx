import { Slider } from '@material-ui/core';
import React from 'react';

const marks = [
  {
    value: 10,
    label: '10'
  },
  {
    value: 12,
    label: '12'
  },
  {
    value: 14,
    label: '14'
  },
  {
    value: 16,
    label: '16'
  }
];

type Props = {
  handleFontChange: (val: number) => void;
  value?: number;
};

export const FontSizeSlider: React.FC<Props> = ({ value, handleFontChange }) => {
  return (
    <Slider
      value={value ?? 14}
      aria-labelledby='discrete-slider-custom'
      step={1}
      onChange={(_, val) => handleFontChange(val as number)}
      valueLabelDisplay='auto'
      marks={marks}
      max={16}
      min={10}
    />
  );
};
