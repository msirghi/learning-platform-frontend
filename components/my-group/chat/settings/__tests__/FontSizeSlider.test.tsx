import { Slider } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { FontSizeSlider } from '../FontSizeSlider';

describe('FontSizeSlider component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<FontSizeSlider handleFontChange={jest.fn()} value={12} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a default value if it was not provided', () => {
    const wrapper = shallow(<FontSizeSlider handleFontChange={jest.fn()} />);
    const slider = wrapper.find(Slider);

    expect(slider).toHaveLength(1);
    expect(slider.props().value).toEqual(14);
  });

  it('should fire prop method on change', () => {
    const spy = jest.fn();
    const wrapper = shallow(<FontSizeSlider handleFontChange={spy} />);
    const slider = wrapper.find(Slider);
    expect(slider).toHaveLength(1);

    slider.props().onChange({} as React.ChangeEvent, 12);
    expect(spy).toBeCalled();
    expect(spy).toBeCalledWith(12);
  });
});
