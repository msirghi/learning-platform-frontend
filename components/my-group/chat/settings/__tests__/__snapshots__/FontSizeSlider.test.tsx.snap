// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontSizeSlider component should match the snapshot 1`] = `
<WithStyles(ForwardRef(Slider))
  aria-labelledby="discrete-slider-custom"
  marks={
    Array [
      Object {
        "label": "10",
        "value": 10,
      },
      Object {
        "label": "12",
        "value": 12,
      },
      Object {
        "label": "14",
        "value": 14,
      },
      Object {
        "label": "16",
        "value": 16,
      },
    ]
  }
  max={16}
  min={10}
  onChange={[Function]}
  step={1}
  value={12}
  valueLabelDisplay="auto"
/>
`;
